### ISSUE TITLE
sprout requires P-CSCF to specify a Path header

### ISSUE THREAD

**Issue by mirw** [Metaswitch/sprout#639]
Tested with a P-CSCF that did not add a Path header (instead, inserting itself in the Contact header).

sprout accepted the REGISTER and stored an entry in the registration store.  However, when it came to route to that binding, it failed, giving the following output in SAS.

```
Starting SIP resolving for address:
Starting NAPTR lookup for target:
The NAPTR lookup failed. Performing SRV lookups for target:  for UDP and TCP
Starting A/AAAA lookup for target:
The A/AAAA lookup has returned for target:
```

From the fact that the hostname is missing from each of these, it looks as though it is trying to route via the non-existent Path (now Route) header.

**Comment by bossmc** [comment]
Raising priority as this is (mostly) valid behaviour for a P-CSCF.

**Comment by rkd-msw** [comment]
I've retested this (by hacking Bono not to add a Path header) and verified that in this case Sprout routes to the address from the REGISTER's Contact header, not a blank address:

```
19-07-2015 11:20:03.957 UTC Debug memcachedstore.cpp:453: Read 437 bytes from table reg key sip:4155550605@rkd.cw-ngv.com, CAS = 570
19-07-2015 11:20:03.957 UTC Debug regstore.cpp:130: Data store returned a record, CAS = 570
19-07-2015 11:20:03.957 UTC Debug regstore.cpp:422: Try to deserialize record for sip:4155550605@rkd.cw-ngv.com with 'JSON' deserializer
19-07-2015 11:20:03.957 UTC Debug regstore.cpp:953: Deserialize JSON document: {"bindings":{"<urn:uuid:d0b1d1cc-823f-50b3-954c-b38ce1c61e61>":{"uri":"sip:quaff@10.0.2.15:37154;transport=TCP;ob","cid":"b82019b0467e523e6020f603d187a1a8","cseq":3,"expires":1437305103,"priority":0,"params":{"+sip.instance":"\"<urn:uuid:d0b1d1cc-823f-50b3-954c-b38ce1c61e61>\""},"paths":[],"timer_id":"00000000000000ef0000000000981000","private_id":"4155550605@rkd.cw-ngv.com","emergency_reg":false}},"subscriptions":{},"notify_cseq":2}
19-07-2015 11:20:03.957 UTC Debug regstore.cpp:978:   Binding: <urn:uuid:d0b1d1cc-823f-50b3-954c-b38ce1c61e61>
19-07-2015 11:20:03.957 UTC Debug regstore.cpp:427: Deserialization suceeded
19-07-2015 11:20:03.957 UTC Debug contact_filtering.cpp:114: Performing contact filtering on binding <urn:uuid:d0b1d1cc-823f-50b3-954c-b38ce1c61e61>
19-07-2015 11:20:03.957 UTC Debug contact_filtering.cpp:338: Trying to match Accept-Contact parameter 'methods' (value 'INVITE')
19-07-2015 11:20:03.957 UTC Debug contact_filtering.cpp:360: Parameter methods is not in the Contact parameters but is not explicitly required
19-07-2015 11:20:03.957 UTC Debug sproutletproxy.cpp:1167: Sproutlet send_request 0x7fb1e855dc60
19-07-2015 11:20:03.957 UTC Verbose sproutletproxy.cpp:1192: scscf-0x7fb1e8559c00 sending Request msg INVITE/cseq=2 (tdta0x7fb1e855d650) on fork 0
19-07-2015 11:20:03.957 UTC Debug sproutletproxy.cpp:1530: Processing actions from sproutlet - 0 responses, 1 requests
19-07-2015 11:20:03.957 UTC Debug sproutletproxy.cpp:1565: Processing request 0x7fb1e855d6f8, fork = 0
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:1683: scscf-0x7fb1e8559c00 transmitting request on fork 0
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:1697: scscf-0x7fb1e8559c00 store reference to non-ACK request Request msg INVITE/cseq=2 (tdta0x7fb1e855d650) on fork 0
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:1522: Removing message 0x7fb1e855dc60 => txdata 0x7fb1e855d6f8 mapping
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:101: Find target Sproutlet for request
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:135: Found next routable URI: sip:quaff@10.0.2.15:37154;transport=TCP;ob
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:260: Found user - quaff
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:260: Found user - quaff
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:260: Found user - quaff
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:260: Found user - quaff
19-07-2015 11:20:03.958 UTC Debug sproutletproxy.cpp:755: No local sproutlet matches request
19-07-2015 11:20:03.958 UTC Debug pjsip: tsx0x7fb1e8578 Transaction created for Request msg INVITE/cseq=2 (tdta0x7fb1e855d650)
19-07-2015 11:20:03.958 UTC Debug basicproxy.cpp:1623: Added trail identifier 74371 to UAC transaction
19-07-2015 11:20:03.958 UTC Debug pjutils.cpp:547: Next hop node is encoded in Request-URI
19-07-2015 11:20:03.958 UTC Debug sipresolver.cpp:86: SIPResolver::resolve for name 10.0.2.15, port 37154, transport 6, family 2
19-07-2015 11:20:03.958 UTC Debug baseresolver.cpp:503: Attempt to parse 10.0.2.15 as IP address
19-07-2015 11:20:03.958 UTC Debug sipresolver.cpp:103: Target is an IP address - default port/transport if required
19-07-2015 11:20:03.958 UTC Info pjutils.cpp:982: Resolved destination URI sip:quaff@10.0.2.15:37154;transport=TCP;ob to 1 servers
19-07-2015 11:20:03.958 UTC Debug pjutils.cpp:547: Next hop node is encoded in Request-URI
19-07-2015 11:20:03.958 UTC Debug basicproxy.cpp:1646: Next hop 10.0.2.15 is not a stateless proxy
```

Looks like some work in this area in the last year (maybe the move to JSON?) has fixed this, so closing.
